accepted/targeting-packs/targeting-packs.md

```diff
@@ -0,0 +1,149 @@
+
+This proposal is about enabling acquisition of the .NET Framework targeting
+packs via NuGet so that compiling code for a specific version of the .NET
+Framework doesn't require a Windows machine with a machine-wide installation of
+the corresponding targeting pack.
+
+Since .NET Framework 2.0, the .NET platform supported targeting different
+versions of the platform regardless of the .NET Framework version that is
+installed on the developer's machine. This allows developers to target a lower
+version than what they use themselves. This is especially useful for in-place
+updates where the older version of the framework cannot be installed side-by-
+side.
+
+That mechanism is called *targeting packs*, which is simply a set of assemblies
+that describe the public APIs of a particular version of the .NET Framework.
+Since those assemblies are only used by the compiler, they don't need to contain
+any code, which we call [reference assemblies](https://www.youtube.com/watch?v=EBpY1UMHDY8).
+
+Unfortunately, the targeting packs aren't available by themselves; they are
+redistributed as part of the .NET Framework Developer Packs which bundle both
+the runtime as well as the targeting pack. This design ensures that developers
+can never be in a state where they have a targeting pack that is higher than the
+.NET Framework version they have installed. That design made sense coming from a
+Visual Studio-first perspective but doesn't serve us well today:
+
+* **It's not cross-platform friendly**. The .NET Framework Developer Packs are
+  only available for Windows which makes the targeting packs indirectly Windows-
+  only too.
+
+* **It's not CI friendly**. Requiring a machine-wide install isn't friendly for
+  cloud-hosted CI machines as developers typically lack permissions to install
+  software.
+
+This document proposes to offer the targeting packs via NuGet that they can be
+acquired in a non-impactful way. It builds on the following existing specs:
+
+* [Phillip Carter](https://github.com/cartermp): [Add Target Packs on NuGet](https://github.com/dotnet/core/pull/64)
+* [Daniel Plaisted](https://github.com/dsplaisted): [Reference Assembly NuGet packages](https://gist.github.com/dsplaisted/83d67bbcff9ec1d0aff1bea1bf4ad79a)
+* [Prototype](https://github.com/dsplaisted/ReferenceAssemblyPackages)
+
+## Scenarios and User Experience
+
+* Build a project which is multi-targeted to .NET Core or .NET Standard as well
+  as .NET Framework on Mac OS or Linux using the .NET CLI
+
+* Build a project targeting Mono (which uses the same Target Framework
+  Identifier as .NET Framework) using the .NET CLI
+
+## Requirements
+
+### Goals
+
+* Doesn't require running an installer
+* Doesn't require Windows
+* Can be used in SDK-style multi-targeting projects
+
+### Non-Goals
+
+* Supporting NuGet-based acquisition of the .NET Framework runtime
+* Supporting NuGet-based acquisition from non-SDK-style projects
```

Comment:
I understand this is a non-goal (and why), but what would prevent this from "just working"?

Reply:
I think we should allow this to work if it's not difficult.  It should be possible with the proof of concept packages I created.

Reply:
I wrote this before NuGet was doubling down on deprecating and replacing `packages.config`. I think fully supporting this in non-SDK style projects is much more viable now.

Reply:
I'd really like to see this supported in old style `.csproj` projects that are using `<PackageReference>`...that's where many of us in ASP.NET will still be for a while yet.

Reply:
@NickCraver The current plan is that it will be supported, you'll just have to explicitly add the `PackageReference`

Reply:
Oh that changes things a bunch and adds other issues.